### ISSUE TITLE
"PRODUCT_NAME was misdefined

### ISSUE THREAD

**Issue by GRyabkin** [bielikb/xcframeworks#7]
"PRODUCT_NAME was misdefined: `MySDK`. Please, provide :product_name option"

Error

**Comment by bielikb** [comment]
Hi @GRyabkin,
could you please provide your project, including the fastlane setup, e.g. in form of a zip? 

Thanks

**Comment by GRyabkin** [comment]
@bielikb I don't think it's required. The error I receive happens inside fastlane action. I duplicated my issue in their repository

**Comment by GRyabkin** [comment]
I have set Product name in build settings, here's my fastlane config:
```
  desc "Export xcframework"
  lane :export_xcframework do
    create_xcframework(
      workspace: 'MySDK.xcworkspace',
      scheme: 'MySDK',
      include_bitcode: false,
      destinations: ['iOS'],
      product_name: "MySDK",
      xcframework_output_directory: 'artifacts/MySDK'
    )
  end
```

**Comment by GRyabkin** [comment]
Ohh, this action builds my `xcframework` in the folder where the action invoked ignoring `xcframework_output_directory`. That's bad

**Comment by bielikb** [comment]
Im owner of both repos, thanks for bringing your issue up. To proceed further I might need to know a little more about your setup.
You can try out the running example right on this repo.
Here's the [lane](https://github.com/bielikb/xcframeworks/blob/72fa34426070543f0d46a5af82768af42b12f31c/fastlane/Fastfile#L4) that creates the xcframework and exports it in the dedicated export folder

**Comment by GRyabkin** [comment]
@bielikb sorry, I can't send you my setup, but I can give you answers to questions if you have them. It's cool that the action works for some basic case

**Comment by GRyabkin** [comment]
I have my Bundle name set as $(PRODUCT_NAME) maybe hardcoding this param helps but it's strange that it's not taken into account. Also, look fastlane action xcodebuild has derivedDataPath parameter I am surprised that you recommend me to cd to folder where derived data will be generated

**Comment by bielikb** [comment]
Is it hard to reproduce your setup in form of a dummy project?

In terms of details, let's step back.

1. What version of the fastlane plugin are you currently using?
2. What version of Xcode IDE are you using?
3. Is the target name that builds your dyamic framework the same as the "product name"?
4. Have you tried to not include the product_name se
4. does the fastlane plugin run fine from the terminal?
5. Could you share the dump of the logs/any additional useful info?

Additionally you're welcome to adjust the functionality of the open sourced [plugin](https://github.com/bielikb/fastlane-plugin-create_xcframework), in that case, please follow the [contribution guideline](https://github.com/bielikb/fastlane-plugin-create_xcframework/blob/master/CONTRIBUTING.md)

**Comment by GRyabkin** [comment]
@bielikb yes, this is a production setup which has many third-party-dependencies, also I have signed NDA.

1. 2.160.0
2. Version 12.0 (12A7209)
3. Yes
4. Yes (the same error)
5. I run it from the terminal always
6. The end looks like this (names are removed):

```
[21:14:20]: ▸ Running script 'Run Script'
[21:14:22]: ▸ Archive Succeeded
+------------------------------+------------------------+
|                     Lane Context                      |
+------------------------------+------------------------+
| DEFAULT_PLATFORM             | ios                    |
| PLATFORM_NAME                | ios                    |
| LANE_NAME                    | ios export_xcframework |
| XCODEBUILD_DERIVED_DATA_PATH |                        |
| XCODEBUILD_ARCHIVE           | 1_MySDK.xcarchive |
+------------------------------+------------------------+
[21:14:22]: ▸ PRODUCT_NAME was misdefined: `MySDK`. Please, provide :product_name option

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying fastlane version          | 0           |
| 2    | default_platform                    | 0           |
| 3    | update_fastlane                     | 0           |
| 4    | Switch to ios preconditions lane    | 0           |
| 5    | cocoapods                           | 19          |
| 6    | (cd ../../my_folder && ./some_script.sh) | 9           |
| 💥   | create_xcframework                  | 764         |
+------+-------------------------------------+-------------+

[21:14:22]: fastlane finished with errors

[!] ▸ PRODUCT_NAME was misdefined: `MySDK`. Please, provide :product_name option
```

**Comment by GRyabkin** [comment]
@bielikb is it something with a default value for config item on line 242?
https://github.com/bielikb/fastlane-plugin-create_xcframework/blob/master/lib/fastlane/plugin/create_xcframework/actions/create_xcframework_action.rb

**Comment by GRyabkin** [comment]
@bielikb also looks like `-archivePath` for `xcodebuild` command doesn't have xcframework_output_directory appended

**Comment by GRyabkin** [comment]
@bielikb the error was generated by line 28 in the [following file](https://github.com/bielikb/fastlane-plugin-create_xcframework/blob/2552b2f2fd6eb409a7e145a352da53b5bf8c0003/lib/fastlane/plugin/create_xcframework/helper/create_xcframework_helper.rb#L24). I am not sure but does it mean I should call my output folder similar to the sample you have?? `Products`?

**Comment by GRyabkin** [comment]
@bielikb I would like to contribute. Do you happen to share some guides on how to test the action without deploying to the Fastlane repository?

**Comment by GRyabkin** [comment]
@bielikb could it be just fixed by returning a value on line 8 of the [file](https://github.com/bielikb/fastlane-plugin-create_xcframework/blob/2552b2f2fd6eb409a7e145a352da53b5bf8c0003/lib/fastlane/plugin/create_xcframework/helper/create_xcframework_helper.rb#L24)

**Comment by GRyabkin** [comment]
@bielikb today I had more time to test out your sample and compare the build settings of your sample and my framework. Well, I have figured out what was the cause of the problem. `Installation directory` of the framework I build was set to `/Applications` but in your sample `$(LOCAL_LIBRARY_DIR)/Frameworks`. This path used by `xcodebuild create_xcframework` to search for resulting binaries for combining.

I am happy I have resolved the issue, I should start to investigate your sample before digging into your code. Sorry, I did it in the late evening.
